rust/src/dataset.rs

```diff
@@ -1063,10 +1077,26 @@ pub(crate) async fn write_manifest_file(
         )
         .await?;
 
+    let id = uuid::Uuid::new_v4().to_string();
+    let staging_path =
+        Path::parse(format!("{path}-{id}")).map_err(|err| CommitError::OtherError(err.into()))?;
     // Then, create a copy at the latest manifest path
+    // While copy is usually atomic in object stores, copy is NOT atomic on *nix
```

Comment:
Should we push this "atomic rename" logic to a ObjectStore trait? And let `ObjectStore` to handle the difference between local posix file system and object stores.

Reply:
https://github.com/lancedb/lance/issues/1202